src/test/java/uk/gov/justice/digital/delius/integration/secure/OffendersResource_getOffenderNsisByCrn.java

```diff
@@ -0,0 +1,113 @@
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.restassured.RestAssured;
+import io.restassured.config.ObjectMapperConfig;
+import io.restassured.config.RestAssuredConfig;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import uk.gov.justice.digital.delius.data.api.NsiWrapper;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles("dev-seed")
+@RunWith(SpringJUnit4ClassRunner.class)
+public class OffendersResource_getOffenderNsisByCrn {
+
+    private static final String OFFENDERS_PATH = "/offenders/crn/%s/convictions/%s/nsis/%s";
+
+    private static final Long KNOWN_CONVICTION_ID = 2500295345L;
+
+    private static final String KNOWN_OFFENDER = "X320741";
+
+    @LocalServerPort
+    private int port;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Value("${test.token.good}")
+    private String validOauthToken;
+
+    @Before
+    public void setup() {
+        RestAssured.port = port;
+        RestAssured.basePath = "/secure";
+        RestAssured.config = RestAssuredConfig.config().objectMapperConfig(
+                new ObjectMapperConfig().jackson2ObjectMapperFactory((aClass, s) -> objectMapper));
+    }
+
+    @Test
+    public void getGetOffenderNsisByCrnAndConvictionId() {
+        final var nsiWrapper = given()
+                .auth()
+                .oauth2(validOauthToken)
+                .contentType(APPLICATION_JSON_VALUE)
+                .when()
+                .get(String.format(OFFENDERS_PATH, KNOWN_OFFENDER, KNOWN_CONVICTION_ID, "BRES,BRE"))
+                .then()
+                .statusCode(HttpStatus.OK.value())
+                .extract()
+                .body()
+                .as(NsiWrapper.class);
+
+        assertThat(nsiWrapper.getNsis()).hasSize(2);
+    }
+
+    @Test
+    public void getGetOffenderNsisByCrnAndConvictionIdLimitCode() {
+        final var nsiWrapper = given()
+            .auth()
+            .oauth2(validOauthToken)
+            .contentType(APPLICATION_JSON_VALUE)
+            .when()
+            .get(String.format(OFFENDERS_PATH, KNOWN_OFFENDER, KNOWN_CONVICTION_ID, "BRES"))
+            .then()
+            .statusCode(HttpStatus.OK.value())
+            .extract()
+            .body()
+            .as(NsiWrapper.class);
+
+        assertThat(nsiWrapper.getNsis()).hasSize(1);
+    }
+
+    @Test
+    public void getGetOffenderNsisByCrnAndUnknownConvictionId_ReturnsEmptyCollection() {
```

Comment:
Further to the earlier comment about the NSI type possibly being a query param - I'd expect this to be a 404 if the conviction does not exist. 

For this URL`/offenders/crn/<some_crn>/convictions/<some_conviction_id>/nsis?nsiTypes=BRE,BRES` that would read to me something like "For the conviction with <some_conviction_id> belonging to the offender with CRN <some_crn> get me any NSIs with types in "BRE,BRES".

If either of the offender or the conviction does not exist that's a hard fail to my mind, but I don't know if there's some existing context I'm missing here?

Reply:
Also agree, though in real world the conviction missing is probably "worse" than a 404 since it almost certainly means a coding error. So 404 is best but return empty list is probably not going to be too bad since we should never see a missing conviction, but may as well be consistent. Certainly shouldn't block the PR being merged.

Reply:
So the first ever ticket  I did in community API we agreed that a missing conviction ID when fetching attendances would give en empty list. I was attempting to maintain consistency